draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
+   in {{protocol-description}}.
+
+   Upon receipt of a QUIC packet that is not of type Initial or 0-RTT,
+   the load balancer extracts as many of the earliest octets from the
+   destination connection ID as necessary to match the server ID length.
+
+   The load balancer decrypts the server ID using 128-bit AES in counter
+   (CTR) mode, much like QUIC packet number decryption. The counter
+   input to AES-CTR is the bytes of the connection ID that do not
+   constitute the encrypted server ID.
+
+   server_id = AES-CTR(key, non-server-id-bytes, encrypted_server_id)
+
+   The output of the decryption is the server ID that the load balancer
+   uses for routing.
+
+### Server Actions
+
+   When generating a routable connection ID, the server writes its
+   provided server ID into the server ID octets, and arbitrary bits
+   into the remaining required connection ID octets. These arbitrary
+   bits MAY encode additional information, but SHOULD appear
+   essentially random to observers.
+
+   The server then encrypts the server ID bytes using 128-bit AES in
+   counter (CTR) mode, much like QUIC packet number decryption. The counter
+   input to AES-CTR is the bytes of the connection ID that do not
+   constitute the encrypted server ID.
+
+   encrypted_server_id = AES-CTR(key, non_server_id_bytes, server-id)
+   
+# Protocol Description {#protocol-description}
+
+   The fundamental protocol requirement is to share the choice of
+   routing algorithm, and the relevant parameters for that algorithm,
+   between load balancer and server.
+
+   For Plaintext CID Routing, this consists of the Routing Bits,
+   Divisor, and Modulus. The Modulus is unique to each server,
+   but the others MUST be global.
+
+   For Encrypted CID Routing, this consists of the Server ID,
+   Server ID Length, Key, and Connection ID Length. The Server ID
+   is unique to each server, but the others MUST be global.
+
+## Out of band sharing
+
+   When there are concerns about the integrity of the path between
+   load balancer and server, operators may share routing information
+   using an out-of-band technique, which is out of the scope of
+   this specification.
+
+   To simplify configuration, the global parameters can be shared
+   out-of-band, while the load balancer sends the unique server
+   IDs via the truncated message formats presented below.
+
+## QUIC-LB Message Exchange
+
+   QUIC-LB load balancers send the encoding parameters to servers
+   as they discover the servers, using a single packet to each that
+   resembles QUIC. They periodically retransmit this packet to each
+   server until that server responds with a QUIC-LB ack. Specifics
+   of this retransmission are implementation-dependent.
+
+   These message formats are specific to QUICv2 and experimental
+   versions leading up to QUICv2. They may require revision for
+   future versions of QUIC.
+
+## Packet Header Format
+   QUIC-LB messages resemble QUIC packets with a long header. They are
```

Comment:
They don't resemble long header packets, they are long header packets, with 0 length CIDs. They follow the invariant design for long headers.